ckan/model/activity.py

```diff
@@ -240,7 +256,19 @@ def package_activity_list(
         q = _filter_activitites_from_type(
             q, include=False, types=exclude_activity_types)
 
-    return _activities_at_offset(q, limit, offset)
+    if after:
+        q = _filter_activitites_from_timestamp(q, timestamp=after, after=True)
+    if before:
+        q = _filter_activitites_from_timestamp(q, timestamp=before, after=False)
+
+    # revert sort queries for "only after" queries
+    revese_order = before and not after
+    results = _activities_limit(q, limit, offset, revese_order).all()
```

Comment:
instead of calling `_activities_limit` please use `order_by`, `offset` and `limit` here. We should aim to remove functions like `_activities_limit` because they make the code harder to understand and optimize.

Reply:
updated [here](https://github.com/avdata99/ckan/commit/64443b263186615ffda09c0cc5c155690d278442)